src/Makefile.am

```diff
@@ -0,0 +1,17 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I don' like the top-level directory name "src" here. To me, when I see that, it implies that all source code goes in src/. Seeing that all under src/ is API, maybe shuffle things up?

Reply:
Same here: When I see `/src` I initially expect non-header source files there (implementations). I see this proposes subtrees under `/src` expressly for different categories of header files.

Reply:
I'm trying to look forward to a future where the source files are moved under this location. E.g. "It provides scope for, over the lifetime of 8.0, restructuring more of the source code in to less insane places". When we don't do that, we get stuck. I was hoping to plan ahead for once and not forbid anything that's not immediately useful.